misk-jdbc/src/main/kotlin/misk/jdbc/DataSourceService.kt

```diff
@@ -72,6 +73,10 @@ class DataSourceService @JvmOverloads constructor(
     hikariConfig.poolName = qualifier.simpleName
     hikariConfig.connectionTimeout = config.connection_timeout.toMillis()
     hikariConfig.validationTimeout = config.validation_timeout.toMillis()
+    hikariConfig.idleTimeout = config.connection_idle_timeout.let {
+      val idleTimeout = it ?: config.connection_max_lifetime.minus(DEFAULT_IDLE_TIMEOUT_OFFSET)
```

Comment:
How would it ever be null in this case within a let block? Or is it shadowing another it variable?

Reply:
Its a `.let` rather than a `?.let`, so it really is an `it: Duration?`. The equivalent would be:
```
    hikariConfig.idleTimeout =
      (config.connection_idle_timeout ?: config.connection_max_lifetime.minus(
        DEFAULT_CONNECTION_IDLE_TIMEOUT_OFFSET
      )).toMillis()
```

Reply:
Yes, I much prefer that final iteration. Thanks!

Reply:
Ah right, I shouldn't code with morning brain 😓